Problem: custom getter function is not necessary

yihuang opened the pull request:
Solution:
- specify the signer tag in proto file

<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰    
v    Before smashing the submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

Closes: #XXX

## Description

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review.
-->

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

yihuang commented on CHANGELOG.md:
```diff
@@ -60,6 +60,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (evm) [#448](https://github.com/crypto-org-chain/ethermint/pull/448) Refactor the evm transfer to be more efficient.
 * (evm) [#450](https://github.com/crypto-org-chain/ethermint/pull/450) Refactor transient stores to be compatible with parallel tx execution.
 * (evm) [#454](https://github.com/crypto-org-chain/ethermint/pull/454) Migrate transient stores to object stores.
+* (evm) [#]() Remove custom signer getter, use the option in protobuf file instead.
```
```suggestion
* (evm) [#461](https://github.com/crypto-org-chain/ethermint/pull/461) Remove custom signer getter, use the option in protobuf file instead.
```

mmsqe commented on cmd/ethermintd/root.go:
```diff
@@ -176,14 +175,6 @@ func NewRootCmd() (*cobra.Command, ethermint.EncodingConfig) {
-	initClientCtx, _ = clientcfg.ReadDefaultValuesFromDefaultClientConfig(initClientCtx)
-	autoCliOpts.Keyring, _ = keyring.NewAutoCLIKeyring(initClientCtx.Keyring)
-	autoCliOpts.ClientCtx = initClientCtx
-	if err := autoCliOpts.EnhanceRootCommand(rootCmd); err != nil {
```
we still need query for modules

yihuang replied:
yeah, no need to revert now.

yihuang commented on go.mod:
```diff
@@ -244,6 +244,8 @@ require (
 	// release/v0.50.x
 	cosmossdk.io/store => github.com/crypto-org-chain/cosmos-sdk/store v0.0.0-20240408014408-56c165536d0a
+	// release/v0.50.x
+	cosmossdk.io/x/tx => github.com/yihuang/cosmos-sdk/x/tx v0.0.0-20240410034106-52679093569e
```
https://github.com/crypto-org-chain/cosmos-sdk/pull/276

Commit:
Problem: custom getter function is not nesserary

Solution:
- specify the signer tag in proto file

add signer tag

Update CHANGELOG.md

Signed-off-by: yihuang <huang@crypto.com>

replace x/tx package

update

merged
